src/lib.rs

```diff
@@ -2,6 +2,9 @@
 #![cfg_attr(feature = "cargo-clippy", allow(new_without_default_derive))]
 #![deny(warnings)]
 
+// Stack type inference requires a deeper recursion limit
+#![recursion_limit="128"]
```

Comment:
I wonder if breaking up the router stack into a series of separate `let` bindings would help with this?

Reply:
this will be removed in a followup